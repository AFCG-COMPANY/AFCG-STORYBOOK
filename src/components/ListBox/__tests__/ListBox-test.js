import React from 'react';
import { mount } from 'enzyme';
import ListBox from '../';

describe('ListBox', () => {
  let mockProps;

  beforeEach(() => {
    mockProps = {
      type: 'default',
      children: <ListBox.Field />,
      className: 'bx--list-box__container',
      disabled: false,
      innerRef: jest.fn(),
    };
  });

  it('should render', () => {
    const wrapper = mount(<ListBox {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an inline class if the type=`inline`', () => {
    const wrapper = mount(<ListBox {...mockProps} type="inline" />);
    expect(wrapper.find('.bx--list-box--inline').length).toBe(1);
  });

  it('should render a disabled class if disabled is true', () => {
    const wrapper = mount(<ListBox {...mockProps} disabled />);
    expect(wrapper.find('.bx--list-box--disabled').length).toBe(1);
  });

  it('should call the provided `innerRef` function with a ref to the node', () => {
    mount(<ListBox {...mockProps} />);
    expect(mockProps.innerRef).toHaveBeenCalled();
  });

  it('should add the provided `className` to the root node', () => {
    const wrapper = mount(<ListBox {...mockProps} />);
    expect(
      wrapper
        .children()
        .prop('className')
        .includes(mockProps.className)
    ).toBe(true);
  });
});
