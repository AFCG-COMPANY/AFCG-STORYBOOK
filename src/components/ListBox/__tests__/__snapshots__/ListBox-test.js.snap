// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBox should render 1`] = `
<ListBox
  className="bx--list-box__container"
  disabled={false}
  innerRef={
    [MockFunction] {
      "calls": Array [
        Array [
          <div
            class="bx--list-box__container bx--list-box"
          >
            <div
              class="bx--list-box__field"
              role="button"
              tabindex="0"
            />
          </div>,
        ],
      ],
    }
  }
  type="default"
>
  <div
    className="bx--list-box__container bx--list-box"
    onKeyDown={[Function]}
  >
    <ListBoxField>
      <div
        className="bx--list-box__field"
        role="button"
        tabIndex="0"
      />
    </ListBoxField>
  </div>
</ListBox>
`;
